File: src/01-easy/retro-typewriter-art/solution.pl
$| = 1;
my %ab = (sp => ' ', bS => '\\', sQ => "'", nl => "\n");
chomp(my $line = <STDIN>);
my $out = '';
for my $tok (split / /, $line) {
    my ($ch, $num);
    my $l2 = length($tok) >= 2 ? substr($tok, -2) : '';
    if (exists $ab{$l2}) { $ch = $ab{$l2}; $num = substr($tok, 0, length($tok)-2); }
    else { $ch = substr($tok, -1); $num = substr($tok, 0, length($tok)-1); }
    my $n = $num eq '' ? 1 : int($num); $n = 1 if $n < 1;
    $out .= $ch x $n;
}
print "$out\n";

File: src/01-easy/retro-typewriter-art/solution.scala
import scala.io.StdIn
object Solution {
    def main(args: Array[String]): Unit = {
        val ab = Map("sp"->' ',"bS"->'\\',"sQ"->'\'',"nl"->'\n')
        val line = StdIn.readLine()
        val sb = new StringBuilder
        for (tok <- line.split(" ")) {
            val l2 = if(tok.length>=2) tok.takeRight(2) else ""
            val (ch, num) = if(ab.contains(l2)) (ab(l2), tok.dropRight(2)) else (tok.last, tok.dropRight(1))
            val n = if(num.isEmpty) 1 else math.max(1, num.toInt)
            sb.append(ch.toString * n)
        }
        println(sb)
    }
}
